[CI] org.elasticsearch.xpack.restart.MlConfigIndexMappingsFullClusterRestartIT > testMlConfigIndexMappingsAfterMigration {cluster=UPGRADED} FAILED

### CI Link

https://gradle-enterprise.elastic.co/s/qzsv5lfrl7m5u/failure?expanded-stacktrace=WyIwIl0#1

### Repro line

`./gradlew ':x-pack:qa:full-cluster-restart:v8.12.0#bwcTest' -Dtests.class="org.elasticsearch.xpack.restart.MlConfigIndexMappingsFullClusterRestartIT" -Dtests.method="testMlConfigIndexMappingsAfterMigration {cluster=UPGRADED}" -Dtests.seed=837504182D6C0AE5 -Dtests.bwc=true -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=it-IT -Dtests.timezone=SystemV/AST4ADT -Druntime.java=21`

### Does it reproduce?

Didn't try

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

```
org.elasticsearch.xpack.restart.MlConfigIndexMappingsFullClusterRestartIT > testMlConfigIndexMappingsAfterMigration {cluster=UPGRADED} FAILED
    java.lang.RuntimeException: An error occurred orchestrating test cluster.
        at __randomizedtesting.SeedInfo.seed([837504182D6C0AE5:87140A6946A9BD88]:0)
        at org.elasticsearch.test.cluster.local.DefaultLocalClusterHandle.execute(DefaultLocalClusterHandle.java:264)
        at org.elasticsearch.test.cluster.local.DefaultLocalClusterHandle.writeUnicastHostsFile(DefaultLocalClusterHandle.java:245)
        at org.elasticsearch.test.cluster.local.DefaultLocalClusterHandle.waitUntilReady(DefaultLocalClusterHandle.java:188)
        at org.elasticsearch.test.cluster.local.DefaultLocalClusterHandle.upgradeToVersion(DefaultLocalClusterHandle.java:161)
        at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster.upgradeToVersion(DefaultLocalElasticsearchCluster.java:150)
        at org.elasticsearch.upgrades.ParameterizedFullClusterRestartTestCase.maybeUpgrade(ParameterizedFullClusterRestartTestCase.java:95)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:980)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
        at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:47)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
        at java.base/java.lang.Thread.run(Thread.java:1583)

        Caused by:
        java.lang.RuntimeException: Elasticsearch process died while waiting for ports file. See console output for details.
            at org.elasticsearch.test.cluster.local.AbstractLocalClusterFactory$Node.lambda$waitUntilReady$0(AbstractLocalClusterFactory.java:274)
            at org.elasticsearch.test.cluster.util.Retry.lambda$retryUntilTrue$0(Retry.java:28)
            ... 1 more
```

<details><summary>Other failure</summary>

```
REPRODUCE WITH: ./gradlew ':x-pack:qa:full-cluster-restart:v8.12.0#bwcTest' -Dtests.class="org.elasticsearch.xpack.restart.FullClusterRestartIT" -Dtests.method="testSingleDoc {cluster=UPGRADED}" -Dtests.seed=837504182D6C0AE5 -Dtests.bwc=true -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=hu-HU -Dtests.timezone=Europe/Sofia -Druntime.java=21

org.elasticsearch.xpack.restart.FullClusterRestartIT > testSingleDoc {cluster=UPGRADED} FAILED
    java.lang.RuntimeException: An error occurred orchestrating test cluster.
        at __randomizedtesting.SeedInfo.seed([837504182D6C0AE5:E900740FC16E51CF]:0)
        at org.elasticsearch.test.cluster.local.DefaultLocalClusterHandle.execute(DefaultLocalClusterHandle.java:264)
        at org.elasticsearch.test.cluster.local.DefaultLocalClusterHandle.writeUnicastHostsFile(DefaultLocalClusterHandle.java:245)
        at org.elasticsearch.test.cluster.local.DefaultLocalClusterHandle.waitUntilReady(DefaultLocalClusterHandle.java:188)
        at org.elasticsearch.test.cluster.local.DefaultLocalClusterHandle.upgradeToVersion(DefaultLocalClusterHandle.java:161)
        at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster.upgradeToVersion(DefaultLocalElasticsearchCluster.java:150)
        at org.elasticsearch.upgrades.ParameterizedFullClusterRestartTestCase.maybeUpgrade(ParameterizedFullClusterRestartTestCase.java:95)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:980)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
        at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:47)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
        at java.base/java.lang.Thread.run(Thread.java:1583)

        Caused by:
        java.lang.RuntimeException: Elasticsearch process died while waiting for ports file. See console output for details.
            at org.elasticsearch.test.cluster.local.AbstractLocalClusterFactory$Node.lambda$waitUntilReady$0(AbstractLocalClusterFactory.java:274)
            at org.elasticsearch.test.cluster.util.Retry.lambda$retryUntilTrue$0(Retry.java:28)
            ... 1 more
```

</details>

I don't think this has anything to do with ML:

```
[2024-01-08T16:36:52,945][ERROR][o.e.b.Elasticsearch      ] [test-cluster-1] fatal exception while booting Elasticsearch
org.elasticsearch.ElasticsearchException: failed to load metadata
        at org.elasticsearch.gateway.GatewayMetaState.start(GatewayMetaState.java:120) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.node.Node.start(Node.java:322) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.bootstrap.Elasticsearch.start(Elasticsearch.java:458) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.bootstrap.Elasticsearch.initPhase3(Elasticsearch.java:251) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:73) ~[elasticsearch-8.13.0.jar:?]
Caused by: org.elasticsearch.gateway.CorruptStateException: org.elasticsearch.xcontent.XContentParseException: [-1:14223] [data_stream] failed to parse field [data_stream]
        at org.elasticsearch.gateway.PersistedClusterStateService.readXContent(PersistedClusterStateService.java:679) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.lambda$loadOnDiskState$6(PersistedClusterStateService.java:578) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.consumeFromType(PersistedClusterStateService.java:721) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.loadOnDiskState(PersistedClusterStateService.java:577) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.loadBestOnDiskState(PersistedClusterStateService.java:501) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.loadBestOnDiskState(PersistedClusterStateService.java:412) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.GatewayMetaState.createOnDiskPersistedState(GatewayMetaState.java:181) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.GatewayMetaState.createPersistedState(GatewayMetaState.java:149) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.GatewayMetaState.start(GatewayMetaState.java:107) ~[elasticsearch-8.13.0.jar:?]
        ... 4 more
Caused by: org.elasticsearch.xcontent.XContentParseException: [-1:14223] [data_stream] failed to parse field [data_stream]
        at org.elasticsearch.xcontent.ObjectParser.throwFailedToParse(ObjectParser.java:616) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:611) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ObjectParser.parseSub(ObjectParser.java:629) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:315) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:166) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.cluster.metadata.DataStreamMetadata.fromXContent(DataStreamMetadata.java:230) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:54) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:147) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:414) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.cluster.metadata.Metadata$Builder.fromXContent(Metadata.java:2686) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.readXContent(PersistedClusterStateService.java:677) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.lambda$loadOnDiskState$6(PersistedClusterStateService.java:578) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.consumeFromType(PersistedClusterStateService.java:721) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.loadOnDiskState(PersistedClusterStateService.java:577) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.loadBestOnDiskState(PersistedClusterStateService.java:501) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.loadBestOnDiskState(PersistedClusterStateService.java:412) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.GatewayMetaState.createOnDiskPersistedState(GatewayMetaState.java:181) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.GatewayMetaState.createPersistedState(GatewayMetaState.java:149) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.GatewayMetaState.start(GatewayMetaState.java:107) ~[elasticsearch-8.13.0.jar:?]
        ... 4 more
Caused by: org.elasticsearch.xcontent.XContentParseException: [-1:14209] [data_stream] unknown field [failure_store]
        at org.elasticsearch.xcontent.ObjectParser.lambda$errorOnUnknown$2(ObjectParser.java:105) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:299) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:166) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.cluster.metadata.DataStream.fromXContent(DataStream.java:976) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.cluster.metadata.DataStreamMetadata.lambda$static$1(DataStreamMetadata.java:65) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ObjectParser.lambda$declareField$10(ObjectParser.java:431) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:609) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ObjectParser.parseSub(ObjectParser.java:629) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:315) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:166) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.cluster.metadata.DataStreamMetadata.fromXContent(DataStreamMetadata.java:230) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:54) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:147) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:414) ~[elasticsearch-x-content-8.13.0.jar:?]
        at org.elasticsearch.cluster.metadata.Metadata$Builder.fromXContent(Metadata.java:2686) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.readXContent(PersistedClusterStateService.java:677) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.lambda$loadOnDiskState$6(PersistedClusterStateService.java:578) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.consumeFromType(PersistedClusterStateService.java:721) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.loadOnDiskState(PersistedClusterStateService.java:577) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.loadBestOnDiskState(PersistedClusterStateService.java:501) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.PersistedClusterStateService.loadBestOnDiskState(PersistedClusterStateService.java:412) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.GatewayMetaState.createOnDiskPersistedState(GatewayMetaState.java:181) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.GatewayMetaState.createPersistedState(GatewayMetaState.java:149) ~[elasticsearch-8.13.0.jar:?]
        at org.elasticsearch.gateway.GatewayMetaState.start(GatewayMetaState.java:107) ~[elasticsearch-8.13.0.jar:?]
        ... 4 more
```

I think it's caused by upgrading from 8.12.0-SNAPSHOT, which understands features behind feature flags, to 8.13.0 (non-snapshot release build), which does not.

--- a/x-pack/qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/xpack/restart/AbstractXpackFullClusterRestartTestCase.java
+++ b/x-pack/qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/xpack/restart/AbstractXpackFullClusterRestartTestCase.java
@@ -36,6 +36,7 @@ public abstract class AbstractXpackFullClusterRestartTestCase extends Parameteri
         .keystore("xpack.watcher.encryption_key", Resource.fromClasspath("system_key"))
         .keystore("xpack.security.transport.ssl.secure_key_passphrase", "testnode")
         .feature(FeatureFlag.TIME_SERIES_MODE)
+        .feature(FeatureFlag.FAILURE_STORE_ENABLED)
         .build();
 
     public AbstractXpackFullClusterRestartTestCase(FullClusterRestartUpgradeStatus upgradeStatus) {

